backstage/backstage
# [Search] Migrate search to new backend system

## Hey, I just made a Pull Request!

part of #13084 

Co-authored-by: Camila Loiola <camilaibs@gmail.com>

In this PR we migrate search plugins following the [migration guide](https://backstage.io/docs/backend-system/building-plugins-and-modules/migrating), to support new backend system https://backstage.io/blog/2023/02/15/backend-system-alpha. 

All new services, extension points and modules are exported from the _alpha_ subpath meaning it can only be imported as the following `import { searchIndexRegistryExtensionPoint } from '@backstage/plugin-search-backend-node/alpha';`

The general recommendation for now is not to migrate any backend installations, but a how to guide is provided to experiment with things. See https://github.com/backstage/backstage/pull/16710/commits/9a034509e134a0994761d863cd5fa5ea3511d97e#diff-9b628643c177e9cc2ef602563fff9caf668efc6185e88386c4efe550e4df0609

### How to test: 

1. disable the permission framework in the app-config

```yaml
permission:
  enabled: false
```

2. spin up `backend-next` (experimental backend used for backend system migrations)
```bash
cd packages/backend-next && yarn start
```

you should see indexing processes kicking off in the logs, just as before

3. spin up app
```bash
yarn start
```


you should be able to navigate to the search plugin and explore search results, just as before.


As part of this migration we have also migrated collators to be its own modules. 

#### :heavy_check_mark: Checklist
